tools/bot/matrix.go

```diff
@@ -0,0 +1,147 @@
+package bot
+
+import (
+	"TUM-Live/tools"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+// Matrix strategy
+type Matrix struct {
+}
+
+// Message represents a Matrix message event that includes html formatting as specified
+// in https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+type Message struct {
```

Comment:
These strategy specific stuff should either be in a matrix package or be renamed: 

```suggestion
type MatrixMessage struct {
```

Reply:
I renamed it, since the name of a struct should not contain the name of the file, but yeah, this might cause conflicts at some point. Calling it `MatrixMessage` should be fine for now, I guess. :+1: